src/main/java/core/basesyntax/service/impl/ReaderServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.ReaderService;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+public class ReaderServiceImpl implements ReaderService {
+    @Override
+    public List<String> readFromFile(String filePath) {
+        List<String> dataFromCsvInput;
+        try {
+            dataFromCsvInput = Files.readAllLines(Path.of(filePath));
+        } catch (IOException e) {
+            throw new RuntimeException(String.format("No such file or directory %s %s",
+                    filePath, e));
```

Comment:
**e** is a second constructor argument, not a msg part. It's important!
```suggestion
            throw new RuntimeException(String.format("No such file or directory %s ",
                    filePath), e);
```

Reply:
> **e** is a second constructor argument, not a msg part. It's important!

Thank you!